src/Suppressor.jl

```diff
@@ -110,6 +110,21 @@ macro suppress_err(block)
     end
 end
 
+function _safe_write_ignore_close_task(out_io, in_io)
+    try
+        write(out_io, in_io)
+    catch
+        # Swallow this exception so the task fails gracefully.
+        # HACK:
+        # Now, create a new sticky task to force the Thread to forget about
+        # this task, which would otherwise keep the IOs alive, since they're
+        # captured in the lambda this task was created with.
+        # (This should really be fixed in the scheduler. See:
+        # https://github.com/JuliaLang/julia/issues/40626)
+        @async nothing
+    end
+end
```

Comment:
😕 okay SO: this _does_ fix the tests, and this is now a working solution.

HOWEVER: It made me realize one other caveat with this approach.
Before, a task could _throw an exception_ if it attempted to log from one thread while another thread is doing `@capture_err`.
NOW: it's possible for its log message to be SILENTLY DROPPED during that race condition, which is maybe just as bad?

So maybe the real answer is actually to fix https://github.com/JuliaLang/julia/issues/47759 on the logger side, and leave this package alone? :/

It's sad because this fix seemed easier, but maybe less so now after these changes.